### PR TITLE
test: use `match` argument in `pytest.raises` to simplify assertions

### PR DESCRIPTION
**Pull request by thiagowfx**
Docs: https://stackoverflow.com/a/56569533/1745064

### CODE REVIEW

File: tests/network/test_remove_intercept.py [hunk 1]
```diff
@@ -21,7 +21,12 @@
 
 @pytest.mark.asyncio
 async def test_remove_intercept_no_such_intercept(websocket):
-    with pytest.raises(Exception) as exception_info:
+    with pytest.raises(
```
Comment (sadym-chromium) [inline tests/network/test_remove_intercept.py:24]: Not related to the PR: This should be `invalid argument`, as `params` are not provided.

Comment (thiagowfx) [reply tests/network/test_remove_intercept.py:24]: They are provided:

```
        await execute_command(
            websocket, {
                "method": "network.removeIntercept",
                "params": {
                    "intercept": "00000000-0000-0000-0000-000000000000",
                },
            })
```

Comment (thiagowfx) [reply tests/network/test_remove_intercept.py:24]: Just expand the diff

File: tests/script/test_shared_id.py [hunk 2]
```diff
@@ -166,7 +167,13 @@ async def test_sharedId_in_different_navigable(websocket, context_id, html):
...
     await goto_url(websocket, context_id, html("some other page"))
 
-    with pytest.raises(Exception) as exception_info:
+    with pytest.raises(
+            Exception,
+            match=re.compile(
+                str({
+                    "error": "no such node",
+                    "message": 'SharedId ".*" belongs to different document. Current document is .*'
+                }))):
```
Comment (sadym-chromium) [inline tests/script/test_shared_id.py:176]: I'd prefer to have a more readable verification here. What about the following?

```suggestion
            check=lambda e: {
                "error": "no such node",
                "message": AnyContains("different document")
            } == exception_info.value.args[0]
        )
```

File: tests/script/test_shared_id.py [hunk 3]
```diff
@@ -209,7 +211,12 @@ async def test_sharedId_not_found(websocket, context_id, html):
 
     shared_id = result["result"]["sharedId"] + "9999"
 
-    with pytest.raises(Exception) as exception_info:
+    with pytest.raises(Exception,
```
Comment (sadym-chromium) [inline tests/script/test_shared_id.py:214]: same here

**Review by sadym-chromium — APPROVED** [review]
LGTM, but consider changing the regex to `check` lambda

File: tests/script/test_shared_id.py (hunk 2 above)
Comment (thiagowfx) [reply tests/script/test_shared_id.py:176]: Where did you get the `check` argument from?

https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest-raises

It doesn't exist

Comment (thiagowfx) [reply tests/script/test_shared_id.py:176]: The only way to do what you're suggesting is to revert to how it was before whenever a regex is involved.

Comment (sadym-chromium) [reply tests/script/test_shared_id.py:176]: Indeed it's not in the API, my bad

**Review by sadym-chromium — APPROVED** [review]

**Pull request merged**